File: contains-duplicate/radiantchoi.swift
class Solution {
    func containsDuplicate(_ nums: [Int]) -> Bool {
        var occurences = [Int: Bool]()

        for num in nums {
            if let occurs = occurences[num] {
                return occurs
            } else {
                occurences[num] = true
            }
        } 

        return false
    }
}

File: product-of-array-except-self/radiantchoi.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        reduced = 1
        zeroes = 0

        for num in nums:
            if num != 0:
                reduced *= num
            else:
                zeroes += 1
        
        result = []

        for num in nums:
            if num == 0:
                result.append(0 if zeroes - 1 > 0 else reduced)
            else:
                result.append(0 if zeroes else reduced // num)
        
        return result
